lib/twitter_cldr/js/mustache/implementation/tokenizers/numbers/rbnf_tokenizer.coffee

```diff
@@ -58,3 +36,32 @@ class TwitterCldr.RBNFTokenizer
   tokenize : (pattern) ->
     tokenizer = new TwitterCldr.PatternTokenizer(null, @tokenizer)
     tokenizer.tokenize(pattern)
+
+  @get_rule_regexp : ->
+    #  Special note about the "rule" token recognizer.
+    #  It needs to match the ruby equivalent of %%?[[:word:]\-]+, i.e., %%? followed by
+    #  any number of '-'s  or characters in one of the following Unicode general categories Letter,
+    #  Mark, Number, Connector_Punctuation. The Javascript version of that has been derived using
+    #  an existing tool available here: http://apps.timwhitlock.info/js/regex
```

Comment:
The cset library referenced here does something very similar to what [RangeSet](https://github.com/twitter/twitter-cldr-rb/blob/master/lib/twitter_cldr/utils/range_set.rb) was designed to accomplish. Stupid UTF-16!